### ISSUE TITLE
[Bug]: set_size only works for loc='center'

### ISSUE THREAD

**Issue by szwang1990** [matplotlib/matplotlib#27772]
### Bug summary

**ax.title.set_size()** works for **loc='center'** but not for **loc='left'** or **'right'**

### Code for reproduction

```python
fig, ax = plt.subplots()
ax.title(label='abc', loc='left')
ax.title.set_size(50)
```

### Actual outcome

![loc_center](https://github.com/matplotlib/matplotlib/assets/37451757/17a63aea-e66a-4d82-9249-026a587e595f)
![loc_left](https://github.com/matplotlib/matplotlib/assets/37451757/05892b13-3c93-4c1f-8656-331cf2efca39)
![loc_right](https://github.com/matplotlib/matplotlib/assets/37451757/9dd51d1a-eb6d-4734-865d-e36a089aff62)

### Expected outcome

**set_size()** works for all title locations ('left', 'center', 'right')

### Operating system

Ubuntu

### Matplotlib Version

3.5.2

### Matplotlib Backend

module://matplotlib_inline.backend_inline

### Python version

Python 3.7.12

### Jupyter version

jupyter lab 0.35.4

### Installation

conda

**Comment by timhoffm** [comment]
To access other titles use

```
ax.get_title(loc="left").set_size(50)
```

**Comment by szwang1990** [comment]
> To access other titles use
> 
> ```
> ax.get_title(loc="left").set_size(50)
> ```

**get_title()** returns a string. How can a string call the **set_size()** function?

My question is that when **loc='center'**, the fontsize can be set properly by
```
ax.title(label='abc', loc='center')
ax.title.set_size(50)
```

when **loc='left'** or **loc='right'**, **set_size(50)** does not work.

**Comment by tacaswell** [comment]
There are actually 3 titles on each axes:

```python
import matplatlib.pyplot as plt
fig, ax = plt.subplots()
ax.set_title('center', loc='center')
ax.set_title('left', loc='left')
ax.set_title('right', loc='right')
```

That is, `loc` controls where which one has the text set not where the text goes.  The attribute `ax.title` is the center one (and predates adding `loc` and the other two).

We don't seem to support getting access to the underlying `Text` objects for the left and right titles (I have a vague memory of discussing this when it came in but someone would have to do the archaeology starting from 917f3aeff95ca5edb3121e9826e09306eee4c126 to be sure).

The best way to set the size of an existing left or right title is probably

```python
kwargs = {}
ax.set_title(ax.get_title(loc='left'), loc='left', **kwargs)
```

which is indeed a bit verbose.

**Comment by szwang1990** [comment]
> That is, `loc`controls where which one has the text set not where the text goes.

This sentence acturally solves my puzzle. Thanks :)

**Comment by timhoffm** [comment]
@szwang1990 sorry my bad, I didn't check my recommendation carefully enough. You're right that `get_title()` only returns strings and thus does not help here.

**Comment by timhoffm** [comment]
@jklymak I don't think that set/get_title_left/right is really much of a help it would shorten @tacaswell's solution from

```
ax.set_title(ax.get_title(loc='left'), loc='left', **kwargs)
```
to an only marginally better
```
ax.set_title_left(ax.get_title_left(), **kwargs)
```
and at the cost of intoducing lots of similar functions.

Depending on the importance of the topic, I could imagine three ways forward:

1) The need for this is rare -> @tacaswell's solution is good enough.
2) We expose `Axes.title` but not the left/right title -> make them public as well (as attributes or properties). - Making them public as well doesn't make the situation much more worse than an already public `Axes.title`.
3) Introduce a separate `Axes.set_title_props()` function.

**Comment by jklymak** [comment]
Yes, I'd misunderstood what `get_title` does.  In my opinion it is a bit silly for it to just return the string in the title.  

I think 2, publicly exposing `title.title_left/right`, is probably fine for this use case.

**Comment by timhoffm** [comment]
> Yes, I'd misunderstood what `get_title` does.

I did too, see above. :roll_eyes: 

> I think 2, publicly exposing `title.title_left/right`, is probably fine for this use case.

The only thing that's bothering me then is the naming ambiguity/imprecision `Axes.title` -> `Text` vs. `Axes.get_title()` -> `str`, which we are going in deeper.

**Comment by szwang1990** [comment]
> @szwang1990 sorry my bad, I didn't check my recommendation carefully enough. You're right that `get_title()` only returns strings and thus does not help here.
> 
> Can you explain your use case for setting title styling (here: size) without setting the title text?

Maybe another `loc` parameter in `set_size()`? So that we can use the following command to set the fontsize of left/center/right titles.
`ax.title.set_size(50, loc='left')`

But this may need lots of work, as @tacaswell said, `ax.title` defaults to the center one.

>  The attribute ax.title is the center one (and predates adding loc and the other two).

**Comment by tacaswell** [comment]
`ax.title` is a `matplotlib.text.Text` object that is an attribute of the `Axes` object and `ax.title.set_size` is an instance method of that `Text` object.  It does not know about the other `Text` objects (which are `ax._left_title` and `ax._right_title` right now).  We definitely do not want all `Text` objects to know how to maybe walk up and the artist tree and set state on a sibling nor do we want to use a special sub-class of `Text` for the title text.

Put differently, `ax.title` _is_ the Python side object that represents the center title.

**Comment by tacaswell** [comment]
> We expose Axes.title but not the left/right title -> make them public as well (as attributes or properties).

My memory of when we added the left/right titles was exactly around this API akwardness and opted to make the private until someone asked for it.

**Comment by tacaswell** [comment]
`ax.left_title_artist`, `ax.center_title_artist`, and `ax.right_title_artist`?

**Comment by timhoffm** [comment]
Checking for consistency, we have a history of exposing `Text` instances:

- `Axis.label`
- `Tick.label1`, `Tick.label2`
- `Axes.title`

So thinking again, I recommend to go with `left_title` / `center_title` / `right_title`. The other variables also don't habe a `_artist` or similar suffix. By making them read-only properties, we clearly error out if somebody tries to assign a string (`ax.center_title = "My title"`). When reading `ax.center_title`, people will realize soon enough that they get a `Text` and not as `str´. That'd be good enough to limit misuse - after all, I'm not aware that somebody complained about the confusion between `str` and `Text` in the above cases.

**Comment by EmployedRussian** [comment]
First, the original problem report had a typo. Instead of `ax.title(label='abc', loc='left')`, what was apparently meant is `ax.set_title(label='abc', loc='left')`.

Second, `ax.set_title(...)` already returns the `Text`, so the fix could be as simple as `ax.set_title(...).set_size(50)`.

Alternatively, this also works:
```
title = ax.set_title(label='abc', loc='left')
title.set_size(50)
```

Both the fact that `set_title` sets one of 3 titles, and that it returns the corresponding title are already [documented](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.set_title.html).

---
That said, `Axes.title` documentation currently [reads](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.html): `title` `str` and links to [set_title](https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.set_title.html#matplotlib.axes.Axes.set_title), which seems wrong -- it's not a `str`, it's a `Text`.

Improving the `Axes.title` documentation to explain that it's a `Text` property for the _center_ title, seems like a logical first step.

Adding `{left,center,right}_title` properties with an explanation that `title` is an alias for `center_title` would be the second.

Does that sound like a reasonable approach?
